File: book-exchange-platform-backend/src/main/java/com/jasmeet/book_exchange_platform/service/BookService.java
package com.jasmeet.book_exchange_platform.service;

import com.jasmeet.book_exchange_platform.dto.CreateBookRequest;
import com.jasmeet.book_exchange_platform.dto.UpdateBookRequest;
import com.jasmeet.book_exchange_platform.entity.Book;
import org.springframework.data.domain.Page;
import org.springframework.web.multipart.MultipartFile;
import com.jasmeet.book_exchange_platform.dto.BookResponse;

import java.util.List;

public interface BookService {

    Book addBookWithImage(
            String title,
            String author,
            String genre,
            String description,
            String bookCondition,
            MultipartFile image
    );

    Book addBook(CreateBookRequest request);

    List<BookResponse> getAllBooks();

    BookResponse getBookById(Long id);

    void deleteBook(Long id);

    List<BookResponse> searchBooks(String title);

    List<Book> filterBooksByGenre(String genre);

    Page<Book> getBooksWithPagination(int page, int size);

    List<BookResponse> getMyBooks();

    BookResponse updateBook(Long id, UpdateBookRequest request);
}

File: book-exchange-platform-frontend/src/services/bookService.js
import API from "../api/axios";

// Get all books
export const getAllBooks = async () => {

    const response = await API.get("/books");

    return response.data;
};

// Search books
export const searchBooks = async (title) => {

    const response = await API.get(
        `/books/search?title=${title}`
    );

    return response.data;
};

export const getBookById = async (id) => {
    const response = await API.get(`/books/${id}`);
    return response.data;
};


export const filterBooksByGenre = async (genre) => {

    const response =
        await API.get(
            `/books/filter?genre=${genre}`
        );

    return response.data;
};

export const getPagedBooks = async (
    page,
    size
) => {

    const response =
        await API.get(
            `/books/paged?page=${page}&size=${size}`
        );

    return response.data;
};

export const getMyBooks = async () => {
    const response = await API.get("/books/my-books");
    return response.data;
};

export const deleteBook = async (id) => {
    const response = await API.delete(`/books/${id}`);
    return response.data;
};

export const updateBook = async (id, bookData) => {
    const response = await API.put(`/books/${id}`, bookData);
    return response.data;
};